ops.h

```diff
@@ -693,15 +712,18 @@ template <size_t k>
 static HWY_NOINLINE HWY_MAYBE_UNUSED std::discrete_distribution<int>
 create_distribution(std::array<float, k>& top_k, float temperature) {
   // re-normalize distribution
-  for (size_t i = 0; i < k; ++i) {
-    top_k[i] = exp(log(top_k[i]) / temperature);
-  }
-  float denominator = 0.0f;
-  for (size_t i = 0; i < k; ++i) {
-    denominator += top_k[i];
-  }
-  denominator = 1.0f / denominator;
-  MulByConst(denominator, top_k.data(), k);
+  namespace hn = hwy::HWY_NAMESPACE;
+  using D = hn::ScalableTag<float>;
+  const D d;
+
+  const auto one = hn::Set(d, 1.0f);
+  const auto temperature_inv = hn::Div(one, hn::Set(d, temperature));
+
+  hn::Transform(d, top_k.data(), top_k.size(),
```

Comment:
And also here :) Either way, we can just compute 1.0f / temperature in scalar code and pass that to the lambda or the proposed TransformWithConstant (hopefully there is a shorter name).